cpp/include/raft/matrix/math.cuh

```diff
@@ -310,8 +324,8 @@ void ratio(
  * @param out: output vector of size n_cols
  * @param stream: cuda stream
  */
-template <typename math_t, typename idx_t>
-void argmax(const math_t* in, int n_rows, int n_cols, idx_t* out, cudaStream_t stream)
+template <typename math_t, typename out_t, typename idx_t = int>
+void argmax(const math_t* in, idx_t n_rows, idx_t n_cols, out_t* out, cudaStream_t stream)
```

Comment:
I redently wrote a quick test case for argmax because I realized it was missing. Let's make sure both of these are being tested. Also, ideally we would be adding new APIs that use mdspan and deprecating the raw pointer APIs where possible.

Reply:
I've closed #877 in favor if #937 but once that's in, this PR should use the new separate argmax header. You can also merge that into your branch if it makes it easier. It's still being reviewed but there shouldn't be any huge API changes (mostly tests and docs)

Reply:
Just a small update. #937 should be merged soon.

Reply:
It's been merged so we now have the argmax broken out into its own header and corresponding test.